### ISSUE TITLE
Few tests fails on mips64 and s390x 

### ISSUE THREAD

**Issue by andypost** [SeasX/SeasLog#292]
PHP 7.3 https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/7268

```
 FAILED TEST SUMMARY
---------------------------------------------------------------------
Check for SeasLog log functions with one parameter. [tests/008.phpt]
Check for SeasLog log functions with two parameters. [tests/009.phpt]
Check for SeasLog log functions with three parameters. [tests/010.phpt]
Check for SeasLog::log() function with two parameters. [tests/011.phpt]
Check for SeasLog::log() function with three parameters. [tests/012.phpt]
Check for SeasLog::log() function with four parameters. [tests/013.phpt]
Check for SeasLog::closeLoggerStream() function. [tests/015.phpt]
Check for SeasLog::analyzerCount() function. [tests/016.phpt]
Check for SeasLog::analyzerDetail() function. [tests/017.phpt]
Check for new SeasLog(). [tests/018.phpt]
```
The failed tests mostly because of log dir and return value of `log()` method

**Comment by andypost** [comment]
List of test data
```
src/SeasLog-2.1.0/tests/012.diff
001+ bool(false)
002+ bool(false)
003+ bool(false)
004+ bool(false)
005+ bool(false)
006+ bool(false)
007+ bool(false)
008+ 
009+ Fatal error: Uncaught Exception: SeasLog Invalid Log File - base_path/default/20200525.log in /mnt/testing/php7-pecl-seaslog/src/SeasLog-2.1.0/tests/012.php:10
010+ Stack trace:
001- bool(true)
002- bool(true)
003- bool(true)
004- bool(true)
005- bool(true)
006- bool(true)
007- bool(true)
008- bool(true)
009- bool(true)
010- bool(true)
011+ #0 /mnt/testing/php7-pecl-seaslog/src/SeasLog-2.1.0/tests/012.php(10): SeasLog::log('EMERGENCY', 'this is a emerg...', Array)
012+ #1 {main}
013+   thrown in /mnt/testing/php7-pecl-seaslog/src/SeasLog-2.1.0/tests/012.php on line 10
src/SeasLog-2.1.0/tests/015.diff
002+ bool(false)
003+ bool(false)
004+ bool(false)
005+ bool(false)
006+ bool(false)
004- bool(true)
009+ bool(false)
005- bool(true)
006- bool(true)
007- bool(true)
008- bool(true)
009- bool(true)
src/SeasLog-2.1.0/tests/018.diff
001+ bool(false)
002+ bool(false)
003+ bool(false)
004+ bool(false)
003- bool(true)
004- bool(true)
005- bool(true)
006- bool(true)
src/SeasLog-2.1.0/tests/017.diff
001+ bool(false)
002+ bool(false)
003+ bool(false)
004+ bool(false)
005+ bool(false)
006+ bool(false)
007+ bool(false)
008+ bool(false)
009+ bool(false)
010+ bool(false)
011+ bool(false)
012+ bool(false)
013+ bool(false)
001- bool(true)
002- bool(true)
003- bool(true)
004- bool(true)
005- bool(true)
006- bool(true)
007- bool(true)
008- bool(true)
009- bool(true)
010- bool(true)
011- bool(true)
012- bool(true)
013- bool(true)
src/SeasLog-2.1.0/tests/016.diff
001+ bool(false)
002+ bool(false)
003+ bool(false)
004+ bool(false)
005+ bool(false)
006+ bool(false)
002- bool(true)
003- bool(true)
004- bool(true)
005- bool(true)
006- bool(true)
007- bool(true)
src/SeasLog-2.1.0/tests/008.diff
001+ bool(false)
002+ bool(false)
003+ bool(false)
004+ bool(false)
005+ bool(false)
006+ bool(false)
007+ bool(false)
008+ 
009+ Fatal error: Uncaught Exception: SeasLog Invalid Log File - base_path/default/20200525.log in /mnt/testing/php7-pecl-seaslog/src/SeasLog-2.1.0/tests/008.php:10
001- bool(true)
002- bool(true)
003- bool(true)
004- bool(true)
005- bool(true)
006- bool(true)
007- bool(true)
008- bool(true)
009- bool(true)
010+ Stack trace:
011+ #0 /mnt/testing/php7-pecl-seaslog/src/SeasLog-2.1.0/tests/008.php(10): SeasLog::emergency('this is a emerg...')
012+ #1 {main}
013+   thrown in /mnt/testing/php7-pecl-seaslog/src/SeasLog-2.1.0/tests/008.php on line 10
src/SeasLog-2.1.0/tests/009.diff
001+ bool(false)
002+ bool(false)
003+ bool(false)
004+ bool(false)
005+ bool(false)
006+ bool(false)
007+ bool(false)
008+ 
009+ Fatal error: Uncaught Exception: SeasLog Invalid Log File - base_path/default/20200525.log in /mnt/testing/php7-pecl-seaslog/src/SeasLog-2.1.0/tests/009.php:10
010+ Stack trace:
001- bool(true)
002- bool(true)
003- bool(true)
004- bool(true)
005- bool(true)
006- bool(true)
007- bool(true)
008- bool(true)
009- bool(true)
010- bool(true)
011+ #0 /mnt/testing/php7-pecl-seaslog/src/SeasLog-2.1.0/tests/009.php(10): SeasLog::emergency('this is a emerg...', Array)
012+ #1 {main}
013+   thrown in /mnt/testing/php7-pecl-seaslog/src/SeasLog-2.1.0/tests/009.php on line 10
src/SeasLog-2.1.0/tests/013.diff
001+ bool(false)
002+ bool(false)
003+ bool(false)
004+ bool(false)
001- bool(true)
002- bool(true)
003- bool(true)
004- bool(true)
src/SeasLog-2.1.0/tests/010.diff
001+ bool(false)
002+ bool(false)
003+ bool(false)
004+ bool(false)
005+ bool(false)
006+ bool(false)
001- bool(true)
002- bool(true)
003- bool(true)
004- bool(true)
005- bool(true)
006- bool(true)
src/SeasLog-2.1.0/tests/011.diff
001+ bool(false)
001- bool(true)
```

**Comment by andypost** [comment]
Getting the list of failures on mips64
```
FAILED TEST SUMMARY
---------------------------------------------------------------------
Check for SeasLog log functions with one parameter. [tests/008.phpt]
Check for SeasLog log functions with two parameters. [tests/009.phpt]
Check for SeasLog log functions with three parameters. [tests/010.phpt]
Check for SeasLog::log() function with two parameters. [tests/011.phpt]
Check for SeasLog::log() function with three parameters. [tests/012.phpt]
Check for SeasLog::log() function with four parameters. [tests/013.phpt]
Check for SeasLog::closeLoggerStream() function. [tests/015.phpt]
Check for SeasLog::analyzerCount() function. [tests/016.phpt]
Check for SeasLog::analyzerDetail() function. [tests/017.phpt]
Check for new SeasLog(). [tests/018.phpt]
```

**Comment by 769344359** [comment]
how do you get the mips64 and s390x platform  device?
i do not have mips64 or s390x device
@andypost

**Comment by andypost** [comment]
This devices contributed to Alpinelinux for building, also s390x has docker image and used in CI
Locally I'm using this Makefile to kickstart https://gist.github.com/andypost/30992c35f1a80146de7294851954777e

**Comment by Neeke** [comment]
Fixed PR #294
